rum-injection-installer/installer/nginx_configurator.go

```diff
@@ -0,0 +1,272 @@
+// Extracts the .so file and places it in the modules path
+func (n *NginxConfigurator) DownloadAndInstallModule(arch string) error {
+	if err := n.getModulesPath(); err != nil {
+		return err
+	}
+
+	moduleURL := fmt.Sprintf("https://github.com/DataDog/nginx-datadog/releases/latest/download/ngx_http_datadog_module-%s-%s.so.tgz", arch, n.Version)
+
+	moduleContent, err := downloadFile(moduleURL)
+	if err != nil {
+		return fmt.Errorf("failed to download module: %v", err)
+	}
+
+	signatureURL := moduleURL + ".asc"
+	signatureContent, err := downloadFile(signatureURL)
+	if err != nil {
+		return fmt.Errorf("failed to download signature: %v", err)
+	}
+
+	publicKeyURL := "https://github.com/DataDog/nginx-datadog/releases/latest/download/pubkey.gpg"
+	publicKeyContent, err := downloadFile(publicKeyURL)
+	if err != nil {
+		return fmt.Errorf("failed to download public key: %v", err)
+	}
+
+	tmpDir, err := os.MkdirTemp("", "nginx-module-verification")
+	if err != nil {
+		return fmt.Errorf("failed to create temp directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	moduleFile := filepath.Join(tmpDir, "module.tgz")
+	signatureFile := filepath.Join(tmpDir, "module.tgz.asc")
+	publicKeyFile := filepath.Join(tmpDir, "pubkey.gpg")
+
+	if err := os.WriteFile(moduleFile, moduleContent, 0644); err != nil {
+		return fmt.Errorf("failed to write module file: %v", err)
+	}
+	if err := os.WriteFile(signatureFile, signatureContent, 0644); err != nil {
+		return fmt.Errorf("failed to write signature file: %v", err)
+	}
+	if err := os.WriteFile(publicKeyFile, publicKeyContent, 0644); err != nil {
+		return fmt.Errorf("failed to write public key file: %v", err)
+	}
+
+	if err := verifySignatureGPG(moduleFile, signatureFile, publicKeyFile); err != nil {
+		return fmt.Errorf("failed to verify module signature: %v", err)
+	}
+
+	if err := extractTarGzFile(moduleFile, n.ModulesPath); err != nil {
+		return fmt.Errorf("failed to extract tgz file: %v", err)
+	}
+
+	fmt.Printf("Module successfully downloaded, verified, and installed in %s\n", n.ModulesPath)
+	return nil
+}
+
+func verifySignatureGPG(moduleFile, signatureFile, publicKeyFile string) error {
+	importCmd := exec.Command("gpg", "--import", publicKeyFile)
+	if output, err := importCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to import public key: %v\nOutput: %s", err, output)
+	}
+
+	verifyCmd := exec.Command("gpg", "--verify", signatureFile, moduleFile)
+	if output, err := verifyCmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("signature verification failed: %v\nOutput: %s", err, output)
+	}
+
+	return nil
+}
+
+// Helper function to download a file
+func downloadFile(url string) ([]byte, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	return io.ReadAll(resp.Body)
+}
+
+// Gets the configuration file from 'nginx -T'
+func (n *NginxConfigurator) getConfigLocation() (string, error) {
```

Comment:
```suggestion
func (n *NginxConfigurator) getDefaultConfigLocation() (string, error) {
```

It reflect better what the function does.